### PR TITLE
Add Authereum integration

### PR DESCRIPTION
**Pull request by cwhinfrey**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -29,6 +31,9 @@ import { SIGN_IN, EDIT_PROFILE, CONFIRM_TRANSACTION } from './modals'
 
 import './App.css'
 
+// Authereum integration
+injectWeb3('rinkeby')
```
Comment (makoto) [inline src/App.js]: Can this be injected only at App.js? I think we need to inject somewhere we can pass the network name (eg: https://github.com/wearekickback/app/blob/dev/src/api/web3.js#L112)

**Comment by hiddentao** [comment]
This needs discussion - all our network detection and resolution work happens in https://github.com/wearekickback/app/blob/dev/src/api/web3.js

Ideally this needs to integrate with that.

File: src/components/Header/AuthereumButton.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import AuthereumLogo from '../../assets/authereum_shield.svg'
+import { GlobalConsumer } from '../../GlobalState'
+
+const Authereum = styled('div')`
+  border-radius: 50%;
+  width: 30px;
+  height: 30px;
+  overflow: hidden;
+  margin-left: -18.5px;
+  margin-bottom: -42.5px;
```
Comment (jefflau) [inline src/components/Header/AuthereumButton.js]: Integers only for pixel values. They'll be rounded anyway so let's make it explicit

File: src/components/Header/AuthereumButton.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  cursor: pointer;
+  padding: 5px;
+  background-color: white;
+  @media (max-width: 576px) {
+    margin-left: -15px;
+    margin-bottom: 0px;
+    margin-top: 17.5px;
```
Comment (jefflau) [inline src/components/Header/AuthereumButton.js]: Integer for pixels

File: src/components/Header/AuthereumButton.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+  async checkForAuthereumUser() {
+    let web3
+    try {
+      web3 = await getWeb3()
+      this.setState({ web3 })
+    } catch (e) {
+      console.log('failed to get web3')
+      this.setState({ isAuthereumUser: false })
+      return
```
Comment (jefflau) [inline src/components/Header/AuthereumButton.js]: Refactor as a functional component with hooks.

File: src/components/Header/AuthereumButton.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }
+
+  render() {
+    if (!this.state.isAuthereumUser) {
+      return null
+    }
+
+    return (
+      <GlobalConsumer>
```
Comment (jefflau) [inline src/components/Header/AuthereumButton.js]: Use a hook to consume the global context `useContext`

Comment (jefflau) [reply src/components/Header/AuthereumButton.js]: https://reactjs.org/docs/hooks-reference.html#usecontext

File: src/components/Header/LogoutButton.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    background: none;
+    border: none;
+  }
+  @media (max-width: 576px) {
+    padding: 0px;
+    text-align: left;
+    padding-top: 10px;
+    padding-bottom: 10px;
+    padding-right: 10px;
+  }
```
Comment (jefflau) [inline src/components/Header/LogoutButton.js]: Use mq.small instead and reverse styles so it works with min-width

File: src/components/Modal/Modal.js [hunk 7]
```diff
@@ -65,21 +64,11 @@ const ModalContent = styled('div')`
...
-      ? `
-    height: auto;
-    width: auto;
-  `
-      : null};
+  height: auto;
+  width: auto;
+  @media (max-width: 576px) {
+    width: 75%;
+  }
```
Comment (jefflau) [inline src/components/Modal/Modal.js]: Use mq.small instead and reverse styles so it works with min-width

File: src/components/Utils/LoadingDots.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (jefflau) [inline src/components/Utils/LoadingDots.js]: Remove in favour of react-emotion

Comment (jefflau) [reply src/components/Utils/LoadingDots.js]: Pretty cool loading component!

File: src/components/svg/authereum.svg [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
+ width="596.000000pt" height="596.000000pt" viewBox="0 0 596.000000 596.000000"
+ preserveAspectRatio="xMidYMid meet">
+<metadata>
+Created by potrace 1.15, written by Peter Selinger 2001-2017
+</metadata>
+<g transform="translate(0.000000,596.000000) scale(0.100000,-0.100000)"
+fill="#FF4C2F" stroke="none">
+<path d="M1870 4964 c-300 -96 -602 -192 -673 -214 l-127 -41 0 -882 c0 -528
+4 -926 10 -992 44 -462 244 -900 578 -1264 76 -83 1282 -1134 1316 -1147 14
```
Comment (jefflau) [inline src/components/svg/authereum.svg]: Run through SVGOMG:

https://jakearchibald.github.io/svgomg/

File: src/components/svg/metamask.svg [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+	.st3{fill:#233447;stroke:#233447;stroke-linecap:round;stroke-linejoin:round;}
+	.st4{fill:#CD6116;stroke:#CD6116;stroke-linecap:round;stroke-linejoin:round;}
+	.st5{fill:#E4751F;stroke:#E4751F;stroke-linecap:round;stroke-linejoin:round;}
+	.st6{fill:#F6851B;stroke:#F6851B;stroke-linecap:round;stroke-linejoin:round;}
+	.st7{fill:#C0AD9E;stroke:#C0AD9E;stroke-linecap:round;stroke-linejoin:round;}
+	.st8{fill:#161616;stroke:#161616;stroke-linecap:round;stroke-linejoin:round;}
+	.st9{fill:#763D16;stroke:#763D16;stroke-linecap:round;stroke-linejoin:round;}
+</style>
+<polygon class="st0" points="274.1,35.5 174.6,109.4 193,65.8 "/>
+<g>
```
Comment (jefflau) [inline src/components/svg/metamask.svg]: Run through https://jakearchibald.github.io/svgomg/

File: src/components/svg/ul.svg [hunk 11]
```diff
@@ -0,0 +1 @@
....680,15.332 258.737 C 15.466 261.919,15.955 263.123,16.881 262.550 C 17.056 262.442,17.200 262.821,17.200 263.394 C 17.200 263.995,17.411 264.408,17.700 264.370 C 19.021 264.199,19.200 264.445,19.200 266.433 C 19.200 267.889,19.342 268.416,19.700 268.287 C 21.086 267.791,22.191 269.649,21.137 270.703 C 20.559 271.281,21.458 272.540,22.235 272.242 C 23.005 271.946,23.720 273.069,23.449 274.149 C 23.224 275.043,23.991 276.275,24.473 275.794 C 24.582 275.685,24.985 275.941,25.368 276.364 C 26.093 277.166,25.687 278.800,24.762 278.800 C 24.537 278.800,24.464 278.980,24.600 279.200 C 24.736 279.420,24.432 279.618,23.924 279.640 C 23.016 279.679,23.017 279.685,24.000 279.979 C 24.550 280.144,25.144 280.153,25.321 279.999 C 25.497 279.845,26.097 279.806,26.653 279.912 C 27.628 280.099,27.969 280.506,28.143 281.691 C 28.188 281.996,28.624 282.674,29.112 283.198 C 29.600 283.722,30.000 284.375,30.000 284.648 C 30.000 284.922,30.540 285.530,31.200 286.000 C 32.055 286.609,32.400 287.184,32.400 287.999 C 32.400 288.882,32.606 289.174,33.300 289.272 C 33.795 289.342,34.378 289.805,34.595 290.300 C 34.812 290.795,35.127 291.200,35.295 291.200 C 35.463 291.200,35.976 291.576,36.436 292.036 L 37.
```
Comment (jefflau) [inline src/components/svg/ul.svg]: Run through https://jakearchibald.github.io/svgomg/

**Review by jefflau — CHANGES_REQUESTED** [review]

File: src/components/Auth/WalletModal.js [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+    let didCloseModal = false
+    while (didCloseModal === false) {
+      // Wait a reasonable amount of time to see if the popup has closed
+      await this.sleep(3000)
+      didCloseModal = await signIn()
+    }
+  }
+  ulInit = async () => {
+    window.sessionStorage.setItem('walletSelection', 'universalLogin')
+    console.log('TODO')
```
Comment (jefflau) [inline src/components/Auth/WalletModal.js]: remove console.log

**Comment by jefflau** [comment]
Thanks for making those changes. Do you mind taking a look at the media queries. I think you've got them the wrong way around as you just switched out `@media(max-width)` with `mq.small`, but our `mq.small` uses min-width.

Can you also take a look at the merge conflicts so we get this merged in?

Thanks

**Comment by jefflau** [comment]
Had a go at merging this and trying to get things working. Had some issues with a module not being found inside the npm package `authereum` so I tried to update to the latest authereum beta, but when I click on logging in, nothing seems to happen.

(last couple commits merged incorrectly so can be removed if you need to)

File: public/_redirects [hunk 13]
```diff
@@ -0,0 +1 @@
+/*    /index.html   200
```
Comment (makoto) [inline public/_redirects]: no new line at the end of file

File: src/assets/authereum_shield.svg [hunk 14]
```diff
@@ -0,0 +1,2 @@
...
....4 92.6 1 99.2 4.4 46.2 24.4 90 57.8 126.4 7.6 8.3 128.2 113.4 131.6 114.7 1.4.5 107.4-89.9 124.8-106.5 30.2-28.7 53.8-70.7 62.2-110.9 4.4-21.2 4.7-29.1 4.4-123.1l-.3-88.4-67.9-21.4c-37.3-11.7-68-21.2-68.2-21.1-.1.2 5.5 11.3 12.6 24.8l12.9 24.5 19.3 6.2c10.6 3.4 26.6 8.5 35.6 11.3l16.3 5.1-.4 81.2c-.3 78-.4 81.6-2.4 90.7-7.6 34.3-23.3 63.4-47.4 88-4.6 4.7-28 25.4-51.9 45.9-23.9 20.6-44.7 38.5-46.2 39.9l-2.6 2.4-45.9-39.5c-56.1-48.4-64.5-56.6-77.9-77.2-10.3-15.9-17.9-33-22.5-50.7-4.5-18-4.9-24.7-4.9-104.9v-75.7l14.3-4.5c7.8-2.5 23.8-7.6 35.5-11.3 11.6-3.7 21.2-7.1 21.2-7.4 0-.3 5.6-11.3 12.5-24.4 6.9-13.2 12.5-24.1 12.5-24.3 0-.6-1.2-.2-56 17.3z"/><path d="M295.4 85.2c-.9 1.8-26.1 49.9-55.9 106.8l-54.2 103.5.8 8.4c1.8 17.2 8.2 35.4 18.6 51.9 3.7 6 4.8 7.1 5.8 6 .6-.7 20.4-38.1 44-83.1 23.5-44.9 43-81.7 43.4-81.7.3 0 20.1 36.4 43.9 80.8 23.8 44.4 43.7 81.5 44.3 82.5 1 1.7 1.2 1.6 2.8-.5 10.3-13.5 18.8-35.8 21.2-55.5l1.2-10-56.8-105.9c-31.2-58.2-56.9-106.1-57.1-106.2-.1-.2-1 1.1-2 3z"/><path d="M270 339.9l-27.5 54.9 15.5 13.5c8.5 7.4 21 18.1 27.7 23.7l12.3 10.2 2.7-2.2c1.6-1.2 13.9-11.9 27.4-23.8l24.7-21.6-27.4-54.8c-15.1-30.1-27.5-54.8-27.7-54.8-.1 0-12.6 24.7-27.7 54.9z"/></g></svg>
```
Comment (makoto) [inline src/assets/authereum_shield.svg]: no new line at the end of file

File: src/components/Header/SignInButton.js [hunk 15]
```diff
@@ -65,6 +73,7 @@ function SignInButton() {
         return loggedIn && userProfile ? (
           <>
             {/* <Notifications>Notification</Notifications> */}
+            {/* <LogoutButton /> */}
```
Comment (makoto) [inline src/components/Header/SignInButton.js]: Is this comment necessary?

File: src/layout/Header.js [hunk 16]
```diff
@@ -73,6 +73,7 @@ function Header({ noMargin, noBackground, positionAbsolute }) {
             <NavLink to="/events">Events</NavLink>
             <NavLink to="/pricing">Pricing</NavLink>
             <GuideDropdown />
+            <LogoutButton />
```
Comment (makoto) [inline src/layout/Header.js]: Screenshot 2019-12-06 at 15 20 01

I think we need a bit of gap between the button and sign in button

File: package.json [hunk 17]
```diff
@@ -12,6 +12,7 @@
     "apollo-link-state": "^0.4.1",
     "apollo-upload-client": "^10.0.0",
     "apollo-utilities": "^1.0.21",
+    "authereum": "0.0.4-beta.55",
```
Comment (makoto) [inline package.json]: When I do `yarn deploy:pr` it fails to install npm module.

```
$yarn deploy:pr
yarn run v1.15.2
$ yarn build:release:kovan && scripts/deployTravisBuildToSurge.sh
$ yarn setup --kovan && yarn build
$ scripts/setup.js --kovan
{
  "DEPLOYER_CONTRACT_ADDRESS": "0xB7465Eef8d3334E1f78Af0382201bd079E3d926B",
  "API_URL": "https://kovan.api.kickback.events",
  "NUM_CONFIRMATIONS": 1,
  "ENV": "kovan",
  "DAI_CONTRACT_ADDRESS": "0x1246Ddaf0885A87D9F3DdF0f08d8B6cc57b68c5f",
  "GIT_COMMIT": "ac82ceb13be9c24f8d226149c29084bbc7aae537",
  "LOGROCKET_TOKEN": "",
  "ROLLBAR_TOKEN": "",
  "BLOCKNATIVE_DAPPID": ""
}
$ react-scripts build
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Creating an optimized production build...
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
Failed to compile.

./src/api/web3.js
Cannot find module: 'authereum'. Make sure this package is installed.

You can install this package by running: yarn add authereum.


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: src/config/index.js [hunk 18]
```diff
@@ -1,5 +1,17 @@
...
+//   ENV: process.env.REACT_APP_ENV,
+//   API_URL: process.env.REACT_APP_API_URL,
+//   NUM_CONFIRMATIONS: process.env.REACT_APP_NUM_CONFIRMATIONS,
+//   GIT_COMMIT: process.env.REACT_APP_GIT_COMMIT,
+//   MIXPANEL_ID: process.env.REACT_APP_MIXPANEL_ID,
+//   LOGROCKET_TOKEN: process.env.REACT_APP_LOGROCKET_TOKEN,
+//   ROLLBAR_TOKEN: process.env.REACT_APP_ROLLBAR_TOKEN,
+//   BLOCKNATIVE_DAPPID: process.env.REACT_APP_BLOCKNATIVE_DAPPID,
+//   DEPLOYER_CONTRACT_ADDRESS: process.env.REACT_APP_DEPLOYER_CONTRACT_ADDRESS
+// }
```
Comment (makoto) [inline src/config/index.js]: Please remove comments if you don't need it.

File: src/components/svg/authereum.svg [hunk 19]
```diff
@@ -0,0 +1,2 @@
...
....4 92.6 1 99.2 4.4 46.2 24.4 90 57.8 126.4 7.6 8.3 128.2 113.4 131.6 114.7 1.4.5 107.4-89.9 124.8-106.5 30.2-28.7 53.8-70.7 62.2-110.9 4.4-21.2 4.7-29.1 4.4-123.1l-.3-88.4-67.9-21.4c-37.3-11.7-68-21.2-68.2-21.1-.1.2 5.5 11.3 12.6 24.8l12.9 24.5 19.3 6.2c10.6 3.4 26.6 8.5 35.6 11.3l16.3 5.1-.4 81.2c-.3 78-.4 81.6-2.4 90.7-7.6 34.3-23.3 63.4-47.4 88-4.6 4.7-28 25.4-51.9 45.9-23.9 20.6-44.7 38.5-46.2 39.9l-2.6 2.4-45.9-39.5c-56.1-48.4-64.5-56.6-77.9-77.2-10.3-15.9-17.9-33-22.5-50.7-4.5-18-4.9-24.7-4.9-104.9v-75.7l14.3-4.5c7.8-2.5 23.8-7.6 35.5-11.3 11.6-3.7 21.2-7.1 21.2-7.4 0-.3 5.6-11.3 12.5-24.4 6.9-13.2 12.5-24.1 12.5-24.3 0-.6-1.2-.2-56 17.3z"/><path d="M295.4 85.2c-.9 1.8-26.1 49.9-55.9 106.8l-54.2 103.5.8 8.4c1.8 17.2 8.2 35.4 18.6 51.9 3.7 6 4.8 7.1 5.8 6 .6-.7 20.4-38.1 44-83.1 23.5-44.9 43-81.7 43.4-81.7.3 0 20.1 36.4 43.9 80.8 23.8 44.4 43.7 81.5 44.3 82.5 1 1.7 1.2 1.6 2.8-.5 10.3-13.5 18.8-35.8 21.2-55.5l1.2-10-56.8-105.9c-31.2-58.2-56.9-106.1-57.1-106.2-.1-.2-1 1.1-2 3z"/><path d="M270 339.9l-27.5 54.9 15.5 13.5c8.5 7.4 21 18.1 27.7 23.7l12.3 10.2 2.7-2.2c1.6-1.2 13.9-11.9 27.4-23.8l24.7-21.6-27.4-54.8c-15.1-30.1-27.5-54.8-27.7-54.8-.1 0-12.6 24.7-27.7 54.9z"/></g></svg>
```
Comment (makoto) [inline src/components/svg/authereum.svg]: no new line at the end of file

**Pull request closed without merging**